nautobot/ipam/tests/test_models.py

```diff
@@ -962,6 +962,78 @@ def setUp(self):
         self.status = Status.objects.get(name="Active")
         self.prefix = Prefix.objects.create(prefix="192.0.2.0/24", status=self.status, namespace=self.namespace)
 
+    def test_get_or_create(self):
+        """Assert `get_or_create` method to permit specifying a namespace as an alternative to a parent prefix."""
+        default_namespace = get_default_namespace()
+        # Use a namespace different from the default namespace which has an ipaddress
+        namespace = (
+            Namespace.objects.exclude(pk=default_namespace.pk).filter(prefixes__ip_addresses__isnull=False).first()
+        )
+        # Creating a new status; because we require a unique object when doing a
+        # `get_or_create` without passing in `host/mask_length`` to avoid receiving multiple IP addresses
+        status = Status.objects.create(name="Test Status 1")
+        ipaddress = IPAddress.objects.filter(parent__namespace=namespace).first()
+        host = ipaddress.host
+        mask_length = ipaddress.mask_length
+
+        ipaddress.status = status
+        ipaddress.save()
+
+        with self.subTest("Assert retrieve"):
+            ip_obj, created = IPAddress.objects.get_or_create(
+                host=host,
+                mask_length=mask_length,
+                namespace=namespace,
+                status=status,
+            )
+            self.assertEqual(ip_obj, ipaddress)
+            self.assertFalse(created)
+
+        with self.subTest(
+            "Assert get_or_create utilizes default namespace when retrieving parent if no namespace is provided"
+        ):
+            ipaddress.parent.namespace = default_namespace
+            ipaddress.parent.save()
+            ip_obj, created = IPAddress.objects.get_or_create(
+                host=host,
+                mask_length=mask_length,
+                status=status,
+            )
+            self.assertEqual(ip_obj, ipaddress)
+            self.assertFalse(created)
+
+        with self.subTest("Assert create"):
+            ipaddress.delete()
+
+            ip_obj, created = IPAddress.objects.get_or_create(
+                host=host,
+                mask_length=mask_length,
+                status=status,
+            )
+            self.assertEqual(ip_obj.host, host)
+            self.assertEqual(ip_obj.mask_length, mask_length)
+            self.assertEqual(ip_obj.parent.namespace, default_namespace)
+            self.assertTrue(created)
+
+        with self.subTest("Assert create explicitly defining a non default namespace"):
+            ip_obj.delete()
+            ip_obj, created = IPAddress.objects.get_or_create(
+                host=host, mask_length=mask_length, status=status, namespace=namespace
+            )
+            expected_parent_for_namespace = Prefix.objects.filter(namespace=namespace).get_closest_parent(
+                cidr=f"{host}/{mask_length}", include_self=True
+            )
```

Comment:
Is it safe to assume that a valid parent prefix exists? What if by chance there was only a single `prefix` in the original `namespace`?

In general I think this test might be cleaner, safer, and clearer if it used bespoke data instead of making assumptions about the data created by the factories.